### DISCUSSION TITLE
Discuss on Mailer Error: SMTP connect() failed.

### DISCUSSION THREAD

**Discussion by Ashvinee12** [PHPMailer/PHPMailer#3059] [Q&A]
require 'MailerVendor/autoload.php';

if($this->EmailType=='S')
{
	//Send SMTP Email 
	//---------------------------------------
	$mailer = new PHPMailer;
	$mailer->IsSMTP();
	$mailer->Host = $this->SMTPHost;
	$mailer->Port = 465; //$this->SMTPPort;
	$mailer->SMTPSecure = 'tls';
	$mailer->SMTPAuth = TRUE;
	$mailer->Username =$this->SMTPUserName;
	$mailer->Password =$this->SMTPPassword;
	$mailer->setFrom($this->SMTPFromEmail, $this->SMTPFromName);
	$mailer->AddAddress($to);
	$mailer->IsHTML(true);
	$mailer->Body = $emailContent;
	$mailer->Subject = $subject;
	if($attatchementPath !=""){$mailer->addAttachment($attatchementPath);}			
	$isEmailSend = $mailer->send();
	
	if(!$mailer->send()) {
		echo 'Message could not be sent.';
		echo 'Mailer Error: ' . $mailer->ErrorInfo;
	} else {
		echo 'Message has been sent';
	}
	
	unset($mailer);
	//----------------------------------------------------  
}

I am using above code for sending an email. but getting below error.
![image](https://github.com/PHPMailer/PHPMailer/assets/124759811/80037b40-80c4-491a-9955-c99648bd4779)

**Comment by Synchro** [comment]
What happened when you followed the link to the documentation in the error message and did what it said?

**Reply by Synchro** [reply]
One obvious thing you're doing wrong (and counter to the docs and examples) is mixing up your encryption settings. This won't work:

```php
$mailer->Port = 465; //$this->SMTPPort;
$mailer->SMTPSecure = 'tls';
```

It should be either `ssl` with port 465, or `tls` with port 587.

**Reply by Ashvinee12** [reply]
I am try with this solution but it is not working. Throwing same error

**Reply by Synchro** [reply]
I hate to repeat myself, but what happened when you followed the link to the documentation in the error message and did what it said?

**Reply by Ashvinee12** [reply]
Yes I follow all steps in link all is fine.

**Reply by Ashvinee12** [reply]
autoload file included

**Reply by Synchro** [reply]
OK, I'm glad you resolved your problem.

**Reply by Ashvinee12** [reply]
Problem not resolved yet. Still getting same error

![image](https://github.com/PHPMailer/PHPMailer/assets/124759811/673a4de8-43f8-4946-85f4-98f55ae37b4c)

**Reply by Synchro** [reply]
OK, so you have not read the docs that cover [this exact issue](https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting#smtp-error-could-not-connect-to-smtp-host).
